packages/core/src/modules/credentials/messages/OfferCredentialMessage.ts

```diff
@@ -63,14 +62,9 @@ export class OfferCredentialMessage extends AgentMessage {
   public get indyCredentialOffer(): CredOffer | null {
     const attachment = this.offerAttachments.find((attachment) => attachment.id === INDY_CREDENTIAL_OFFER_ATTACHMENT_ID)
 
-    // Return null if attachment is not found
-    if (!attachment?.data?.base64) {
-      return null
-    }
-
     // Extract credential offer from attachment
-    const credentialOfferJson = JsonEncoder.fromBase64(attachment.data.base64)
+    const credentialOfferJson = attachment?.data?.getDataAsJson()
 
-    return credentialOfferJson
+    return credentialOfferJson as CredOffer
```

Comment:
Same here. Seems to happen in a couple of places. If something changes, could you change it everywhere?

Reply:
as above - added this to match the type. Maybe we can remove null from the fct signature  as getDataAsJson actually throws an error now if it can't provide data. Again, I'm pretty new to typescript. Thought this would result either in an error or undefined but not null the way it is now.  Any help appreciated

Reply:
or we use try catch ?